lib/TokenManager.js

```diff
@@ -47,12 +51,8 @@ function setRefreshTimeout(sdk, tokenMgmtRef, storage, key, token) {
     refreshWait = 0;
   }
   var refreshTimeout = setTimeout(function() {
-    if (tokenMgmtRef.autoRefresh) {
-      return refresh(sdk, tokenMgmtRef, storage, key);
-    } else if (token.expiresAt * 1000 <= Date.now()) {
-      remove(tokenMgmtRef, storage, key);
-      emitExpired(tokenMgmtRef, key, token);
-    }
+    emitExpired(tokenMgmtRef, key, token);
+    return getAsync(sdk, tokenMgmtRef, storage, key);
```

Comment:
I’ve seen the story both ways: some folks want the SPA to be “logged out immediately” on “session end”, some want it to only happen on a state transition or API request.  The former I’ve seen much more, though.

In this area I’m seeing folks get confused about access token expiration vs. okta session expiration.  These are the points I’m finding that we need to clarify for developers:

* The access tokens are minted for and scoped to their applications
* The Okta session is meant as a single-sign-on convenience that spans applications
* You **can** use the Okta session to refresh your app-specific access tokens, until the Okta session expires

To support the “logged out immediately” on “session end” story (where session = Okta session), I believe we should instruct the developer to watch for the token expiration events, and each time a token expires they can query `session.get()` to see if the session is still valid.

Also, IMO we still need the internal “is refreshing” state so that multiple calls to `await token.get()` won’t trigger multiple token-minting calls.  This situation is likely in any application that hasn’t refactored their “token getting” into a service, where they are using direct calls to our library in multiple components that may be rendering at the same time.